hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/balance/AccountBalancesFileLoader.java

```
@@ -110,48 +128,52 @@ public boolean loadAccountBalances(@NonNull File balanceFile, DateRangeFilter da
+                }
             }
 
-            insertedCount += tryInsertBatchAccountBalance(insertBalanceStatement, accountBalanceList, 1);
-            complete = (insertedCount == validCount);
-            updateAccountBalanceSet(updateSetStatement, complete, consensusTimestamp);
-        } catch (InvalidDatasetException | SQLException ex) {
-            log.error("Failed to load account balances file " + fileName, ex);
-        }
+            if (!skip) {
+                insertedCount += tryInsertBatchAccountBalance(insertBalanceStatement, accountBalanceList, 1);
+                complete = (insertedCount == validCount);
+            } else {
+                complete = true;
+            }
 
-        if (complete) {
-            log.info("Successfully processed account balances file {} with {} out of {} records inserted in {}",
-                    fileName, insertedCount, validCount, stopwatch);
-        } else {
-            log.error("ERRORS processing account balances file {} with {} out of {} records inserted in {}", fileName,
-                    insertedCount, validCount, stopwatch);
+            updateAccountBalanceSet(updateSetStatement, complete, consensusTimestamp);
+            updateAccountBalanceFile(updateAccountBalanceFileStatement, consensusTimestamp, validCount,
+                    startTime.getEpochSecond(), Instant.now().getEpochSecond(), fileName);
+
+            if (complete) {
+                log.info("Successfully processed account balances file {} with {} out of {} records inserted in {}",
+                        fileName, insertedCount, validCount, stopwatch);
+            } else {
+                log.error("ERRORS processing account balances file {} with {} records parsed so far in {}",
+                        fileName, validCount, stopwatch);
+            }
+        } finally {
+            DataSourceUtils.releaseConnection(connection, dataSource);
         }
-
-        return complete;
```

This looks like a change in behavior. Previously if not complete then it would not move the file and continue to retry it. Now it is skipping it. We should probably throw an exception above, which causes a rollback and will retry the file later.

---

refactored. after changing `tryInsertBatchAccountBalance` to throw under SQL execution failure, the original condition `complete = (insertedCount == validCount)` would always be true so unnecessary

Now in `loadAccountBalances`, any db error would eventually cause `ParserSQLException` so spring will roll back the transaction